test/smoke/src/areas/positron/editor/fast-execution.test.ts

```diff
@@ -35,7 +33,27 @@ export function setup(logger: Logger) {
 
 				await app.workbench.quickaccess.openFile(join(app.workspacePathOrFolder, 'workspaces', 'fast-statement-execution', FILENAME));
 
-				for (let i = 1; i < 12; i++) {
+				let previousTop = -1;
+
+				// Note that this outer loop iterates 10 times.  This is because the length of the
+				// file fast-execution.r is 10 lines.  We want to be sure to send a Control+Enter
+				// for every line of the file
+				for (let i = 1; i < 11; i++) {
+					let currentTop = await app.workbench.positronEditor.getCurrentLineTop();
+					let retries = 10;
+
+					// Note that top is a measurement of the distance from the top of the editor
+					// to the top of the current line.  By monitoring the top value, we can determine
+					// if the editor is advancing to the next line.  Without this check, the test
+					// would send Control+Enter many times to the first line of the file and not
+					// perform the desired test.
+					while (currentTop === previousTop && retries > 0) {
+						currentTop = await app.workbench.positronEditor.getCurrentLineTop();
+						retries--;
+					}
```

Comment:
What happens if all the retries are exhausted and the top hasn't changed? Should this count as a test fail because it will technically invalidate the rest of the results?

Reply:
The test will fail if the retries are exhausted and the top hasn't changed because a duplicate Control-Enter will be sent to a line that has already received one and the variable validations will be wrong.  Either multiple will fail if a line is skipped or just the last one will fail if the second to the last line receives two Control-Enters.